src/n2n.c

```diff
@@ -312,9 +312,11 @@ int supernode2sock (n2n_sock_t *sn, const n2n_sn_name_t addrIn) {
     return rv;
 }
 
-
+#if defined(_MSC_VER)
+DWORD WINAPI resolve_thread(LPVOID p) {
```

Comment:
I would like to only see feature-depending `ifdefs` (e.g. `N2N_HAVE_PTHREAD`) and avoid platform / compiler specific `ifdef`s (`WIN32` etc.) in the code and rather see the latter in the header `.h` files. Could we please solve it with macro definitions, e.g. change the function definition to a more generic

 ```C
 N2N_THREAD_RETURN_DATATYPE resolve_thread (N2N_THREAD_PARAMETER_DATATYPE p) {
```

and define the two macros in let's say `n2n.h` or `n2n_define.h` or `win32/n2n_win32.h` either to 

```C
#define N2N_THREAD_RETURN_DATATYPE        void*
#define N2N_THREAD_PARAMETER_DATATYPE     void*
```

or 

```C
#define N2N_THREAD_RETURN_DATATYPE       DWORD WINAPI
#define N2N_THREAD_PARAMETER_DATATYPE    LPVOID
```

depending on if `_MSC_VER_` is defined or not?

Reply:
ok, thanks.  update it

Reply:
:+1: